core/App/ehsm_provider.cpp

```diff
@@ -938,7 +940,24 @@ ehsm_status_t GenerateQuote(ehsm_data_t *quote)
     return EH_OK;
 }
 
+/*
+ *  @brief get mrenclave and mrsigner from quote
+ *  @param quote_data quote data
+ *  @param mr_signer the mrsigner
+ *  @param mr_enclave the mrenclave
+ */
+void get_mrenclave_mrsigner_from_quote(const uint8_t* quote_data, 
+                                       uint8_t* mr_signer, 
+                                       uint8_t* mr_enclave)
+{
+    sgx_quote3_t *p_sgx_quote = (sgx_quote3_t *)quote_data;
```

Comment:
change the return code as bool.

you also need to check the cass of the quote is NULL or failed to get the expected fileds.

Reply:
Ok, I'll change as you said.